# Olist Brazilian E-Commerce Data Analysis

## 1. Data Understanding

In [70]:
import pandas as pd
list_of_tables=["C:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/olist_customers_dataset.csv",
"c:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/olist_geolocation_dataset.csv",
"c:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/olist_order_items_dataset.csv",
"c:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/olist_order_payments_dataset.csv",
"c:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/olist_order_reviews_dataset.csv",
"c:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/olist_orders_dataset.csv",
"c:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/olist_products_dataset.csv",
"c:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/olist_sellers_dataset.csv",
"c:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/product_category_name_translation.csv"]
name_table=["olist_customers","olist_geolocation","olist_order_items","olist_order_payments","olist_order_reviews"
,"olist_orders","olist_products","olist_sellers","product_category_name_translation"]
summary=[]
for i in range(len(list_of_tables)):
    df=pd.read_csv(list_of_tables[i])
    table_info={
        "table_name":name_table[i],
        "rows":df.shape[0],
        "columns":df.shape[1],
        "total_null":df.isnull().sum().sum(),
        "duplicates":df.duplicated().sum()
    }
    summary.append(table_info)
table_sumary=pd.DataFrame(summary)
Column_Level_Summary=[]
for i in range(len(list_of_tables)):
    df=pd.read_csv(list_of_tables[i])
    list_columns=list(df.columns)
    for j in range(df.shape[1]):
        column_info={
            "Table":name_table[i],
            "Columns":list_columns[j],
            "Data_Type":df[list_columns[j]].dtype,
            "Nulls":df[list_columns[j]].isnull().sum(),
            "Null_Percentage":((df[list_columns[j]].isnull().sum())/(df.shape[0]))*100,
            "unique":df[list_columns[j]].nunique()
        }
        Column_Level_Summary.append(column_info)
table_columns=pd.DataFrame(Column_Level_Summary)    
print(f"{table_sumary}\n---------------------\n")    
print(table_columns)






                          table_name     rows  columns  total_null  duplicates
0                    olist_customers    99441        5           0           0
1                  olist_geolocation  1000163        5           0      261831
2                  olist_order_items   112650        7           0           0
3               olist_order_payments   103886        5           0           0
4                olist_order_reviews    99224        7      145903           0
5                       olist_orders    99441        8        4908           0
6                     olist_products    32951        9        2448           0
7                      olist_sellers     3095        4           0           0
8  product_category_name_translation       71        2           0           0
---------------------

                                Table                        Columns  \
0                     olist_customers                    customer_id   
1                     olist_customers      

In [77]:
df1=pd.read_csv("c:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/olist_orders_dataset.csv")
df2=pd.read_csv("C:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/olist_customers_dataset.csv")
set_col1=set(df1["customer_id"])
set_col2=set(df2["customer_id"])
is_foreign_key=set_col1.issubset(set_col2)
print(f"is it foreign key ? {is_foreign_key}")


is it foreign key ? True


## 3.1 Duplicate Analysis
### Geolocation Duplicate Investigation

In [ ]:
df=pd.read_csv("c:/Users/Adora IT/Desktop/olist/olist-data-analysis/data/olist_geolocation_dataset.csv")
print(df.duplicated().sum())
print(df['geolocation_zip_code_prefix'].nunique())
info=df.groupby("geolocation_zip_code_prefix").agg(
    records=('geolocation_zip_code_prefix','count'),
    unique_lat=('geolocation_lat','nunique'),
    unique_lng=('geolocation_lng', 'nunique') 

).reset_index()
print(info)


261831
19015
       geolocation_zip_code_prefix  records  unique_lat  unique_lng
0                             1001       26          10          10
1                             1002       13           6           6
2                             1003       17          10          10
3                             1004       22          14          14
4                             1005       25          11          11
...                            ...      ...         ...         ...
19010                        99960        5           5           5
19011                        99965        6           6           6
19012                        99970       21          16          16
19013                        99980       26          21          21
19014                        99990        2           2           2

[19015 rows x 4 columns]


In [93]:
count_without_duplicates = df.drop_duplicates(inplace=False).shape[0]
print(count_without_duplicates)

738332


Geolocation Duplicate Analysis:
The geolocation table contains 1,000,163 records and 19,015 unique ZIP code prefixes. A ZIP code prefix can correspond to multiple geographic coordinates, so repeated ZIP prefixes are not considered duplicates by themselves. However, 261,831 complete duplicate rows were identified. Removing these complete duplicates would leave 738,332 unique records. These duplicates will be removed during the data-cleaning phase